File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.swift
import Foundation

func solution(_ numbers:[Int]) -> String {
    var stringNum = numbers.map {String($0)}
    stringNum.sort {
        return $0 + $1 > $1 + $0
    }
    let answer = stringNum.joined()
    return answer.first == "0" ? "0" : answer
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def solution(numbers):
    
    def compare(a,b):
        if a+b > b+a:
            return -1
        elif a+b < b+a:
            return 1
        else:
            return 0
    
    numbers = list(map(str, numbers))
    numbers.sort(key=cmp_to_key(compare))
    answer = "".join(numbers)
    
    return "0" if answer[0] == "0" else answer